### PR TITLE
Replace proto.MarshalTextString with proto.Marshal

### LINKED ISSUE #8314: Replace proto.MarshalTextString with proto.Marshal
In the process of unMarshal using the String generated by proto.MarshalTextString, the target structure cannot add or delete fields compared to the original structure. In addition, the binary data generated by proto.Marshal can be partially unmarshaled.

### PR DESCRIPTION
**Pull request by czs007**
Resolves: #8314 
Signed-off-by: zhenshan.cao <zhenshan.cao@zilliz.com>

### CODE REVIEW

File: internal/datacoord/meta.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ func (m *meta) reloadFromKV() error {
 	for _, value := range values {
 		segmentInfo := &datapb.SegmentInfo{}
 		// TODO deprecate all proto text marshal/unmarsahl
```
Comment (xiaofan-luan) [inline internal/datacoord/meta.go:60]: Remove this TODO

**Review by xiaofan-luan — APPROVED** [review]

**Review by XuanYang-cn — APPROVED** [review]

**Pull request merged**